service-base/src/main/java/org/eclipse/hono/service/VertxBasedHealthCheckServer.java

```diff
@@ -120,6 +121,15 @@ public void setAdditionalResources(final List<Handler<Router>> resourceProviders
         final HttpServerOptions options = new HttpServerOptions()
                 .setPort(config.getHealthCheckPort())
                 .setHost(config.getHealthCheckBindAddress());
+
+        if (config.isHealthCheckUseSsl()) {
+            options.setSsl(true);
+            final JksOptions keyOptions = new JksOptions();
```

Comment:
It would be great if we could re-use the methods we already have for the key/cert material, and allow for PKS12 and PEM. And not hardcode JKS.

Reply:
True. I will have a look

Reply:
Comming back to this:
At the moment the healthCheck related properties are set as children of `app` for example in the MQTT config, which is extended with the SSL properties:
```yaml
hono:
  app:
    maxInstances: 1
    healthCheckPort: ${vertx.health.port}
    healthCheckBindAddress: 0.0.0.0
    healthCheckUseSsl: true
    healthCheckKeyStorePath: "/etc/hono/auth-server-internal.p12"
    healthCheckKeyStorePassword: "auth-server-internal"
```
To make it work with different key store formats I suggest to move the healthCheck to its own property like so:
```yaml
hono:
  app:
    maxInstances: 1
  healthCheck:
    port: ${vertx.health.port}
    bindAddress: 0.0.0.0
    useSsl: true
    keyStorePath: "/etc/hono/auth-server.p12"
    keyStorePassword: "password"
```

Then we could introduce a new config `HealthCheckConfig` which would extend [`AbstractConfig`](https://github.com/eclipse/hono/blob/master/core/src/main/java/org/eclipse/hono/config/AbstractConfig.java).
With this in place setting the options would go something like that:
```java
if (healthCheckConfig.isUseSsl()) {
  options.setSsl(true);
  options.setKeyCertOptions(healthCheckConfig.getKeyCertOptions());
}
```

WDYT?